R/createFigure4.R

```diff
@@ -0,0 +1,220 @@
+  
+  if('statecode' %in% names(params)){
+    params$statecode = paste0("US:",params$statecode)
+  }
+  
+  starts = seq(as.POSIXct(params$startDateLo), as.POSIXct(params$startDateHi), by=stride)
+  
+  if(length(starts) < 2){
+    ends   = as.POSIXct(params$startDateHi)
+  } else {
+    ends   = c(starts[2:(length(starts))], as.POSIXct(params$startDateHi))
+  }
+  
+  starts = format(starts, '%Y-%m-%d')
+  ends   = format(ends, '%Y-%m-%d')
+  
+  out = list()
+  
+  for(i in seq_along(starts)){
+    cat('Downloading ', starts[i], ' to ', ends[i], '\n')
+    params$startDateLo = starts[i]
+    params$startDateHi = ends[i]
+    
+    if(starts[[i]] != ends[[i]]){
+      chunk.call = do.call(readWQPdata, params)
+    
+      if(!is.null(chunk.call)){
+        meta <- attr(chunk.call, "siteInfo") %>%
+          select(MonitoringLocationIdentifier, dec_lat_va, dec_lon_va, StateCode)
+        
+        chunk.call <- rename(chunk.call, 
+                             Date=ActivityStartDate, 
+                             value=ResultMeasureValue, 
+                             units=ResultMeasure.MeasureUnitCode) %>% 
+          left_join(meta, by="MonitoringLocationIdentifier") %>%
+          rename(wqx.id=MonitoringLocationIdentifier) %>%
+          mutate(StateCode = as.character(StateCode),
+                 ActivityTypeCode = as.character(ActivityTypeCode),
+                 ActivityMediaName = as.character(ActivityMediaName),
+                 ResultStatusIdentifier = as.character(ResultStatusIdentifier)) %>%
+          select(Date, value, units, wqx.id, 
+                 dec_lat_va, dec_lon_va, 
+                 StateCode, 
+                 ActivityTypeCode,ActivityMediaName,
+                 ResultStatusIdentifier)
+        
+        out[[i]] <- chunk.call       
+      }
+    }
+  }
+  
+  if(length(out) > 0){
+    out.data <- bind_rows(out)
+    
+  } else {
+    out.data <- data.frame(Date = numeric(),
+                           value=numeric(),
+                           units = character(),
+                           wqx.id = character(),
+                           dec_lat_va = numeric(),
+                           dec_lon_va = numeric(),
+                           StateCode = character(),
+                           ActivityTypeCode = character(),
+                           ActivityMediaName = character(),
+                           ResultStatusIdentifier = character())
+  }
+  
+  return(out.data)
+
+}
+
+
+
+library(dplyr)
+library(dataRetrieval)
+library(lubridate)
+library(ggplot2)
+
+characteristicNames = c("Depth, Secchi disk depth", 
+                        "Depth, Secchi disk depth (choice list)", 
+                        "Secchi Reading Condition (choice list)", 
+                        "Secchi depth", 
+                        "Water transparency, Secchi disc")
+
+# To get the full data set used to produce the figure in the text:
+# This takes roughly 15 minutes
+get_us_secchi(outfile="all_secchi_usa_long.rds",
```

Comment:
Ah OK. I'm leaving at 5, but everything else works for me now. So once you sort that out: :+1: